internal/cmd/environment/command_delete.go

```diff
@@ -28,7 +28,10 @@ func (c *command) newDeleteCommand() *cobra.Command {
 }
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
-	id := args[0]
+	id, err := convertNameToId(args[0], c.AuthenticatedCLICommand.V2Client)
```

Comment:
What if `args[0]` is already an ID? We shouldn't call the list API in that case.

Reply:
The `delete` commands will need to be updated after @sgagniere merges #1789

Reply:
We have no idea if the user is passing in an ID or a name, so we would have to call the list API no matter what.

Reply:
For the `delete` commands I can just not make any changes to them atm.

Reply:
> We have no idea if the user is passing in an ID or a name, so we would have to call the list API no matter what.

I think all resources either have an ID with a fixed prefix, or else the name itself is the ID (like `topic`). For the former, we can just parse the prefix to see if it's a valid ID for that resource, and for the latter we don't have to do anything.

Reply:
Should we call the describe endpoint first? What I'm worried about is the worst case: certain customers can have thousands of a specific resource, (such as 10,000 ACLs) and calling the list endpoint by default might mean 100 paginated calls instead of just 1 call 😬

Reply:
> We have no idea if the user is passing in an ID or a name, so we would have to call the list API no matter what.

I think all resources either have an ID with a fixed prefix, or else the name itself is the ID (like `topic`). For the former, we can just parse the prefix to see if it's a valid ID for that resource, and for the latter we don't have to do anything.

Edit: Sorry if you get pinged multiple times. Is Github being weird for anyone else? I canceled my pending review because Brian already pointed out the thing I was going to point out, and Github just posted my review anyway 🙄

Reply:
Yeah, from what Steven said, I think if the user passes an ID, since it's prefixed, I can check for that and avoid the list API call. I'll do that then. Thanks for pointing that out!

Reply:
> > We have no idea if the user is passing in an ID or a name, so we would have to call the list API no matter what.
> 
> I think all resources either have an ID with a fixed prefix, or else the name itself is the ID (like `topic`). For the former, we can just parse the prefix to see if it's a valid ID for that resource, and for the latter we don't have to do anything.
> 
> Edit: Sorry if you get pinged multiple times. Is Github being weird for anyone else? I canceled my pending review because Brian already pointed out the thing I was going to point out, and Github just posted my review anyway 🙄

Resources where the name itself is the ID will be unaffected by this PR

Reply:
We already have some support for this, too:
```
if resource.LookupType(serviceAccountId) != resource.ServiceAccount {
```

Reply:
> I think if the user passes an ID, since it's prefixed, I can check for that and avoid the list API call.

What if the user has a Kafka cluster named "lkc-my-kafka-cluster"?

Reply:
> > I think if the user passes an ID, since it's prefixed, I can check for that and avoid the list API call.
> 
> What if the user has a Kafka cluster named "lkc-my-kafka-cluster"?

I'm not sure if we can handle that case unless there's some other rule about IDs after their prefix. If a user were to just name their resource like an ID, we wouldn't have a good way of handling that case (i.e. cluster id: lkc-1234, name: lkc-5678). If there was another cluster with the ID: lkc-5678, and I run `describe lkc-5678`, which one should run? Should we instead place limitations on what users can name their resources (i.e. just don't let them mirror the ID format)?

Reply:
> Should we instead place limitations on what users can name their resources (i.e. just don't let them mirror the ID format)?
I'm not sure what the best path here is, but this we can't do. There's nothing stopping a user from making these names in the UI, and it's probably not good to introduce new naming limitations anyway.

Reply:
Maybe we should always try to run a command with the assumption of the user using an ID, but if it trying the request with the supposed ID fails, we then can treat it as a name, try to find its corresponding ID, and if that still fails, return the error to the user.